include/openthread/trel.h

```diff
@@ -73,6 +73,15 @@ typedef struct otTrelPeer
  */
 typedef uint16_t otTrelPeerIterator;
 
+/**
+ * Enables or disables TREL operation.
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aEnabled   A boolean to enable/disable the TREL operation.
+ *
+ */
+void otTrelSetEnabled(otInstance *aInstance, bool aEnable);
```

Comment:
We already have `otTrelEnable()` and `otTrelDisable()` which does the same.
(and they document the behavior in more detail).

Not sure if we need a new flavor of existing APIs. 
(it will not decrease/help with code size and can lead to confusion for users of API).

Reply:
Should we replace the Enable/Disable with a single SetEnabled()? I think we tend to use SetEnabled() in other places.

Reply:
In general as a policy I think we should try to avoid removing existing public APIs unless we have a good reason/justification. This helps keep the APIs backward compatible. Same with changing existing APIs (e.g., adding new parameters, etc). We can always add new public APIs.

In the case here:
- I am not sure if we have good justification for removing the `Enable/Disable()` APIs.
     - We may argue that these APIs are not widely used and may be fine to change now, but I think it is better to not set a precedence.
- We can add `SetEnabled()` along with `Enable/Disable()` but then 
    -  It can be confusing (why we have multiple APIs doing similar thing and what is the difference) and 
    - I am not sure if it does help simplify the code much (we move an `if` check from caller side to the API side).

Reply:
While I agree we should try not to remove existing API, I think we should also attempt to maintain consistent models. I believe the rest of the OT user APIs all follow the `SetEnabled()` model. Several years ago, we debated `Enable()`/`Disable()` vs. `SetEnabled()` and settled on the latter. Unfortunately, I missed this when reviewing the TREL APIs.

Reply:
I am ok if we decide to remove `Enable/Disable` and replace it with `SetEnabled` (for consistency).

Then we need to add a parallel PR in `ot-br-posix` to use the new API (along with this PR).

Reply:
Thanks for the discussion, if we agree to replace `Enable/Disable` with `SetEnabled`, I plan to do it in 3 steps:
* This PR to add `SetEnabled`
* ot-br-posix PR to switch over to `SetEnabled`
* Remove `Enable/Disable` from openthread
Or are there better ways?

Reply:
Yes. I think this can work.